packages/vscode-extension/src/handlers.ts

```diff
@@ -1944,15 +1948,69 @@ export async function openAdaptiveCardExt(
       "Cancel"
     );
     if (selection === "Install") {
+      ExtTelemetry.sendTelemetryEvent(
+        TelemetryEvent.AdaptiveCardPreviewerInstallConfirm,
+        getTriggerFromProperty(args)
+      );
       await vscode.commands.executeCommand("workbench.extensions.installExtension", acExtId);
-      await vscode.commands.executeCommand("workbench.view.extension.cardLists");
+    } else {
+      ExtTelemetry.sendTelemetryEvent(
+        TelemetryEvent.AdaptiveCardPreviewerInstallCancel,
+        getTriggerFromProperty(args)
+      );
     }
-  } else {
-    await vscode.commands.executeCommand("workbench.view.extension.cardLists");
   }
   return Promise.resolve(ok(null));
 }
 
+function isAdaptiveCard(content: string): boolean {
+  const pattern = /"type"\s*:\s*"AdaptiveCard"/;
+  return pattern.test(content);
+}
+
+export async function hasAdaptiveCardInWorkspace(): Promise<boolean> {
+  // Teams message limit is 28K.
+  // Skip large files which are unlikely to be adaptive cards to prevent performance impact.
+  const fileSizeLimit = (28 + 10) * 1024;
```

Comment:
Please ensure the 28k size limit is reasonable for all projects, for example, Tab / MetaOS app (Outlook / Office)

Reply:
no clear limits from card docs, increased to 1M for safety